src/gfx_device/gfx_device.rs

```diff
@@ -0,0 +1,134 @@
+        match self.gfx_device_inner {
+            GfxDeviceInner::OpenGL { ref window, .. } => window.get_inner_size(),
+            #[cfg(windows)]
+            GfxDeviceInner::Direct3D {} => unimplemented!(),
+            GfxDeviceInner::Null => None,
+        }
+    }
+
+    pub fn render_world(&mut self, world: &mut ecs::World, pipeline: &renderer::Pipeline) {
+        use components::rendering::{MeshInner, TextureInner, Renderable};
+        use components::transform::LocalTransform;
+        use world_resources::camera::Projection;
+        match self.gfx_device_inner {
+            GfxDeviceInner::OpenGL { ref mut renderer,
+                                     ref mut device,
+                                     ref window,
+                                     .. } => {
+                let camera = world.read_resource::<world_resources::Camera>().clone();
+
+                let projection_mat = match camera.projection {
+                    Projection::Perspective {
+                        fov,
+                        aspect_ratio,
+                        near,
+                        far,
+                    } => renderer::Camera::perspective(fov, aspect_ratio, near, far),
+                    Projection::Orthographic {
+                        left,
+                        right,
+                        bottom,
+                        top,
+                        near,
+                        far,
+                    } => renderer::Camera::orthographic(left, right, bottom, top, near, far),
+                };
+                let eye = camera.eye;
+                let target = camera.target;
+                let up = camera.up;
+                let view_mat = renderer::Camera::look_at(eye, target, up);
+                let camera = renderer::Camera::new(projection_mat, view_mat);
+
+                let mut scene = renderer::Scene::<gfx_device_gl::Resources>::new(camera);
+                let renderables = world.read::<Renderable>();
+                let local_transforms = world.read::<LocalTransform>();
+                for (renderable, local_transform) in (&renderables, &local_transforms).iter() {
+                    let (buffer, slice) = match renderable.mesh.mesh_inner {
+                        MeshInner::OpenGL { ref buffer,
+                                            ref slice } => { (buffer.clone(), slice.clone()) },
+                        _ => continue,
+                        };
+                    let ka = match renderable.ka.texture_inner {
+                        TextureInner::OpenGL { ref texture } => texture.clone(),
+                        _ => continue,
+                    };
+                    let kd = match renderable.kd.texture_inner {
+                        TextureInner::OpenGL { ref texture } => texture.clone(),
+                        _ => continue,
+                    };
+                    let transform = local_transform.matrix();
```

Comment:
These should just use the `Transform` component instead of the `LocalTransform` as well. Then there is no need for recalculation of the matrix every frame if isn't changed and there is also the possibility of static lights by just adding a `Transform` without a `LocalTransform`.